lib/encryption.js

```diff
@@ -1,21 +1,25 @@
 const crypto = require('crypto');
-const algorithm = 'aes-256-cbc';
+const ALGORITHM = 'aes-256-cbc';
+const IV_SIZE_BYTES = 16;
+const KEY_SIZE_BYTES = 32;
 
-module.exports = secret => ({
+module.exports = secret => {
+  const iv = crypto.randomBytes(IV_SIZE_BYTES);
 
-  encrypt: data => {
-    const text = JSON.stringify(data);
-    const cipher = crypto.createCipher(algorithm, secret);
-    let crypted = cipher.update(text, 'utf8', 'hex');
-    crypted += cipher.final('hex');
-    return crypted;
-  },
+  // key size for aes-256-cbc must be 32 bytes, pad with random bytes when necessary
+  const key = Buffer.concat([Buffer.from(secret), crypto.randomBytes(KEY_SIZE_BYTES)], KEY_SIZE_BYTES);
```

Comment:
Not sure about the key padding - might be better just to throw an exception if the key length isn't suitable?

Reply:
I think this approach will be generally problematic (also the way that the IV key is randomly generated) because it will be non-deterministic across instances and restarts. That means a session created on one node in a cluster won't be readable on other nodes in the cluster.

All keys will need to be either provided, or generated from provided parameters in a deterministic way.